etc/testdata/baas_server_test_config.json

```diff
@@ -0,0 +1,144 @@
+{
```

Comment:
was this existing? not sure we want all of these vars/pws in our repo.

Reply:
this server config exists in the old `realm-cli`, and its used to spin up a baas server in evergreen.  All the details here are more or less the same as the details you'd find in `etc/configs/local_config.json` (and similar files) that are committed in the `baas` repo.  Caveat being that I copy-pasted this, but I don't _think_ any of these values are truly secret (or meaningful) in any way

Reply:
my only hesitation / concern is baas is internal to us and this is public facing.

Reply:
a fair hesitation!  this has always been a bit of an issue though: the fact that realm-cli is open and baas isn't.

a benefit of doing things this way is it allows us to test against a custom, deterministic Realm server (one spun up locally alongside tests), and not have to write tests against `realm-dev.mongodb.com` (and subsequently worry about creating garbage test-data in those envs)

Reply:
although not as fluent as this solution, can we add an env var or a flag to pass in to point to the baas repo's local config? if we find it's too inconvenient, we can add a Makefile that is git ignored with the intent of local engineers updating it with their location. Thoughts on that?

Reply:
I don't love the idea of having to reach into another repo for this, but don't have many great alternatives for keeping the structure of this JSON document secret.  its cumbersome for sure, but it's also only used/referenced in the `.evg.yml` script (side question: why do we commit `.evg.yml` in public repos, that seems to also expose a lot of info/details?).

the usage is here:
```
          cat > atlas_config.json <<EOF
          {
            "admin": {
              "authProviders": [
                {
                  "name": "mongodb-cloud",
                  "type": "mongodb-cloud",
                  "config": {
                      "baseUrl": "https://cloud-dev.mongodb.com",
                      "adminUsername": "${baas_mongodb_cloud_atlas_admin_username}",
                      "adminAPIKey": "${baas_mongodb_cloud_atlas_admin_api_key}"
                    },
                    "metadataFields": [{ "name": "name", "required": false}]
                }
              ]
            }
          }
          EOF
          go run cmd/server/main.go --configFile $ROOT_DIR/realm-cli/etc/testdata/baas_server_test_config.json --configFile ./atlas_config.json
```

at that point in the evergreen script, we've already cloned the `baas` repo, so if we copied over this config (im not positive its actually a dupe of an existing one yet, but can figure that out), that last command _should_ be able to be `--configFile $ROOT_DIR/baas/etc/configs/cli_test_config.json --configFile ./atlas_config.json`.

honestly, its still a bit of a hack job when running this locally, to this point I've been running the server with `local_cloud_dev_config.json` and then executing tests with:

```cmd
REALM_SERVER_BASE_URL=http://localhost:8080 REALM_MONGODB_CLOUD_ADMIN_USERNAME=tezzrgzr REALM_MONGODB_CLOUD_ADMIN_API_KEY=********-****-****-****-0923ac86a00a go test -v -tags debug github.com/10gen/realm-cli/internal/cloud/realm -run 'Test'
```

(the reason being b/c I can't that evg approach with merging an in-place config for configuring the `mongodb-cloud` auth provider without the evg env vars for username/password)

path of least resistance here seems to be porting this config over to `baas` as `cli_test_config.json` (or finding its duplicate), merging that in, then updating this script to point to that file and removing it here.  thoughts?

Reply:
FYI: will follow up on the rest of this with this ticket: https://jira.mongodb.org/browse/REALMC-7449